tests/cbmc/stubs/s2n_stuffer_read_expected_str.c

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ * this file except in compliance with the License. A copy of the License is
+ * located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "api/s2n.h"
+#include "stuffer/s2n_stuffer.h"
+#include "utils/s2n_safety.h"
+
+#include <cbmc_proof/nondet.h>
+
+int s2n_stuffer_read_expected_str(struct s2n_stuffer *s2n_stuffer, const char *expected)
+{
+    PRECONDITION_POSIX(s2n_stuffer_is_valid(s2n_stuffer));
+    notnull_check(expected);
+    return nondet_bool() ? S2N_SUCCESS : S2N_FAILURE;
```

Comment:
Doesn't this function also advance the read cursor?

Reply:
I worry that not advancing the cursor could mean missing memory safety issues get hidden.

Reply:
By checking the valid of the stuffer on each function, wouldn’t that make sure we will be safe along the way?

Reply:
Lets add an issue to fix that, and a comment on the function warning that its an incomplete stub.

Reply:
I created the issue https://github.com/awslabs/s2n/issues/2173 to track this request and I added the comments.